### ISSUE TITLE
Running tldr with a non-existent topic or doing an update causes long freezes

### ISSUE THREAD

**Issue by sdondley** [tldr-pages/tldr-node-client#441]
I'm on alpine, installed with npm. If I do `tldr blah` I see something like this:

```
✔ Updating...
⠼ Creating index...
```

It just hangs for like two minutes. Same thing happens when `tldr --update` is run.

**Comment by agnivade** [comment]
Yes, this is due to the single threaded nature of tldr. We try to yield in intervals but it's difficult to make the spinner run more smoothly. But it's just due to the addition of the spinner that you "feel" it's stuck. It's just creating the index.

Is there a real bug that you wanted to report?

**Comment by sdondley** [comment]
So you're saying it's not a bug that if I make a typo or type in a non-existent document I have to open a new terminal shell and or wait two minutes for the program to re-index everything? This isn't just a "feeling" I have. 2 minutes is a long time for the command line to lockup and do nothing.

**Comment by agnivade** [comment]
Sorry I realize that my earlier post came across as snarky. Wasn't my intention.

> if I make a typo or type in a non-existent document I have to open a new terminal shell and or wait two minutes for the program to re-index everything? 

You could just cancel the re-indexing. You don't need to wait. The fact that we decide to re-populate the cache if a command isn't found was a design decision based on the fact that the newly entered command might exist in the remote repo, but not in the local cache.

We could, technically add an escape hatch to prevent this behavior. I am open to it. But the core issue here is that indexing takes some time and freezes the UI. And that's not a bug. It's just due to the single threaded design of node.js. I am sure it can be improved. PRs would be most welcome.

**Comment by spageektti** [comment]
What about re-indexing only if page exists in the remote repo (ping https://raw.githubusercontent.com/tldr-pages/tldr/main/pages.xx/platform/page.md)?

**Comment by agnivade** [comment]
Sure, a reasonable approach.

**Comment by kbdharun** [comment]
We have been providing a pre-built index for a while upstream at https://github.com/tldr-pages/tldr/releases/latest/download/index.json, would it be possible for the Node client to be adapted to use it?

**Comment by agnivade** [comment]
The client builds its own search index. There's some logic here: https://github.com/tldr-pages/tldr-node-client/blob/main/lib/search.js. Unless we pre-build this entire logic, it might be difficult.
